File: HISTORY - CODE/Comatose v1.2.1 [UNRELEASED]/Python3/pyComa/pickup.py
from symbol import symbol

class pickup(object):
    def __init__(self, ploc):
        self.loc = ploc
    def printp(self, board):
        for k in self.loc:
            #if ( board.size() < static_cast<whole>(ploc_.at(k)) )
            board[k].set(symbol.P)
    def pickingup(self, bombs, locOFh):
        ''' pickingup
        * - human lands on pickup
        returns new bombs count
        '''
        for n, i in enumerate(self.loc):
            if locOFh == i:
                del self.loc[n]
                return bombs + 10

File: HISTORY - CODE/Comatose v1.0 (comatose3.zip) r3/C++/pickup.cpp
#include "pickup.h"

pickup::pickup(const VINT& ploc)
{
	ploc_.clear();
	for (CNT w = 0; w < ploc.size(); w++)
	{
		ploc_.push_back(ploc.at(w));
	}
}

pickup::~pickup()
{
}

void pickup::printp(SYMBOLS board)
{
	for (CNT k = 0; k < ploc_.size(); k++)
	{
		//if ( board.size() < static_cast<whole>(ploc_.at(k)) )
		{
			board->at(ploc_.at(k)) = symbol::P;
		}
	}
}

/* pickingup
* - human lands on pickup
*/
void pickup::pickingup(CNT& bombs, const whole& locOFh)
{
	for (CNT i = 0; i < ploc_.size(); i++)
	{
		if (locOFh == ploc_.at(i))
		{
			bombs += 10;
			ploc_.erase(ploc_.begin()+i);
			return;
		}
	}
	/*for (vector<int>::iterator it = ploc_.begin(); it < ploc_.end(); it++)
	{
	if (locOFh == *it)
	{
	bombs += 10;
	ploc_.erase(it);
	return;
	}
	}*/
}
